sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -72,24 +95,15 @@ public InputStream toStream() {
     }
 
     /**
-     * Provides {@link Mono} of {@link InputStream} for the data represented by this {@link BinaryData} object.
-     *
-     * @return {@link InputStream} representation of the {@link BinaryData}.
-     */
-    public Mono<InputStream> toStreamAsync() {
-        return Mono.fromCallable(() -> toStream());
-    }
-
-    /**
-     * Create {@link BinaryData} instance with given {@link InputStream} as source of data. The {@link InputStream} is
-     * not closed by this function.
+     * Creates a {@link BinaryData} instance with given {@link InputStream} as source of data. The {@link InputStream}
+     * is not closed by this function.
      *
      * <p><strong>Create an instance from InputStream</strong></p>
      * {@codesnippet com.azure.core.experimental.util.BinaryDocument.from#Stream}
      *
      * @param inputStream to read bytes from.
      * @throws UncheckedIOException If any error in reading from {@link InputStream}.
-     * @throws NullPointerException if {@code inputStream} is null.
+     * @throws NullPointerException If {@code inputStream} is null.
```

Comment:
There feels like an inconsistency in how you handle null or empty data. Sometimes you turn it into an empty BinaryData, other times you throw an exception. Is there a reason for this?

Reply:
I was treating stream/Flux separately. But now , they will behave same.